chutney/engine/src/test/java/com/chutneytesting/engine/domain/execution/strategies/RetryWithTimeOutStrategyTest.java

```diff
@@ -317,4 +316,97 @@ private void stopExecution(ScenarioExecution scenarioExecution) {
+        StepStrategyDefinition strategyDefinition = new StepStrategyDefinition("", strategyProperties);
+
+        Step step = mockStep(Status.SUCCESS);
+        when(step.strategy()).thenReturn(Optional.of(strategyDefinition));
+
+        StepDataEvaluator evaluator = mock(StepDataEvaluator.class);
+        when(step.dataEvaluator()).thenReturn(evaluator);
+        when(evaluator.evaluate(eq("${#RETRY_DELAY}"), any())).thenReturn("100 ms");
+
+        strategyUnderTest.execute(ScenarioExecution.createScenarioExecution(null), step, new ScenarioContextImpl(), null);
+
+        ArgumentCaptor<String> infoCaptor = ArgumentCaptor.forClass(String.class);
+        verify(step, atLeastOnce()).addInformation(infoCaptor.capture());
+
+        List<String> capturedInfos = infoCaptor.getAllValues();
+        assertThat(capturedInfos).anyMatch(info ->
+            info.contains("Retry strategy definition : [timeOut 10 s] [delay 100 ms]"));
+    }
+
+    @Test
+    public void should_evaluate_complex_spel_expressions() {
+        StrategyProperties strategyProperties = properties("${#BASE_TIMEOUT + ' s'}", "${#RETRY_DELAY * 2 + ' ms'}");
+        StepStrategyDefinition strategyDefinition = new StepStrategyDefinition("", strategyProperties);
+
+        Step step = mockStep(Status.SUCCESS);
+        when(step.strategy()).thenReturn(Optional.of(strategyDefinition));
+
+        StepDataEvaluator evaluator = mock(StepDataEvaluator.class);
+        when(step.dataEvaluator()).thenReturn(evaluator);
+        when(evaluator.evaluate(eq("${#BASE_TIMEOUT + ' s'}"), any())).thenReturn("15 s");
+        when(evaluator.evaluate(eq("${#RETRY_DELAY * 2 + ' ms'}"), any())).thenReturn("200 ms");
+
+        strategyUnderTest.execute(ScenarioExecution.createScenarioExecution(null), step, new ScenarioContextImpl(), null);
+
+        ArgumentCaptor<String> infoCaptor = ArgumentCaptor.forClass(String.class);
+        verify(step, atLeastOnce()).addInformation(infoCaptor.capture());
+
+        List<String> capturedInfos = infoCaptor.getAllValues();
+        assertThat(capturedInfos).anyMatch(info ->
+            info.contains("Retry strategy definition : [timeOut 15 s] [delay 200 ms]"));
+    }
+
+    @Test
+    public void should_use_original_value_when_not_spel_expression() {
+        StrategyProperties strategyProperties = properties("20 s", "150 ms");
+        StepStrategyDefinition strategyDefinition = new StepStrategyDefinition("", strategyProperties);
+
+        Step step = mockStep(Status.SUCCESS);
+        when(step.strategy()).thenReturn(Optional.of(strategyDefinition));
+        when(step.dataEvaluator()).thenReturn(new StepDataEvaluator(null));
+
+        strategyUnderTest.execute(ScenarioExecution.createScenarioExecution(null), step, new ScenarioContextImpl(), null);
+
+        verify(step).addInformation(contains("Retry strategy definition : [timeOut 20 s] [delay 150 ms]"));
+    }
+
+    @Test
+    public void should_handle_mixed_spel_and_non_spel_values() {
```

Comment:
It can be removed because this is already checked in `should_evaluate_spel_expression_in_retryDelay `and `should_evaluate_spel_expression_in_timeout`